### ISSUE TITLE
Fetch remaining non-indexed fields of a document directly with txn.Get()

### ISSUE THREAD

**Issue by islamaliev** [sourcenetwork/defradb#1936]
At the moment fetching with secondary indexes is implemented in the following way:
first we fetch the indexed field, extract the doc's docKey and with it using `DocumentFetcher` and `Spans` fetch the remaining field.

Another reasonable approach to consider would be to fetch all the remaining fields directly with `txn.Get()` as we know already the docKey and can construct `core.DatastoreKey` for each field and fetch them.
We still will have to perform filtering.

For this we could implement another fetcher `FieldsFetcher` that would only fetch a single document and run the filter on it.

This can give us some performance boost and we should measure it to make sure it really has an impact. Otherwise it just adds complexity and should be adandoned.

**Comment by jsimnz** [comment]
I don't think this is necessary, although the fetcher currently over fetches, that will be adjusted/fixed in the future.

I'd like to keep doc state fetching within the fetcher, instead of implement the same or similar logic in various places.

If the fetcher currently isn't working for a particular use case, it should be fixed, not avoided.

Having the dockey of a document doesn't prohibit the use of the fetcher, you can use it to narrow your search for the fetcher via spans.

**Comment by islamaliev** [comment]
@jsimnz I agree that we shouldn't create another fetcher with similar functionality but maybe it makes sense to make this ticket as part of `DocumentFether` refactoring endeavour? At least to consider to possible optimisation
